### DISCUSSION TITLE
Recurring Stripe payment should ideally create Customer and Subscription in Stripe only when payment is made

### DISCUSSION THREAD

**Discussion by wei2go** [solspace/craft-freeform#1315] [Questions & Answers] [answered]
### What happened?

While trying out Freeform's Stripe integration, I found that it already create an empty Customer record in Stripe as soon as the page is loaded (if we put name and Stripe payment field on the same page). 

For monthly payment, a similar thing happens to subscriptions. It creates empty 'active' subscriptions (with $0.00) when page is loaded, if we put email field and Stripe payment field on the same page. When the email field is filled, it creates an 'incomplete' subscription, and will be updated to 'active' if the payment is successful later on.

These behaviours create bunches of empty / orphaned customers and subscriptions in Stripe, which is unexpected. The corresponding Customer and Subscription records should only be created when the payment is successfully made.

### How can we reproduce this?

1. Create a form with name, email and Stripe payments (configured as recurring) on the same page
2. Render the form on a front end page
3. Open browser and load the front end page
4. We can then see empty customers and subscriptions in Stripe

### Freeform Edition

Pro

### Freeform Version

5.3.0

### Craft Version

Craft Pro 5.1.7

### When did this issue start?

Fresh install of Freeform

**Comment by kjmartens** [comment, marked as answer]
Hi @wei2go,

I understand your concern and frustration. However, this is out of our control due to Stripe's expectations of how this approach will work. Essentially, they believe that any presentation of the Stripe payment fields is the beginning of a legitimate interaction. For this reason, they recommend that you place the Stripe payment fields on the last page of a multipage form. For example, you might collect user information on page 1, payment amount information on page 2, and then present the payment field on page 3 (or pages 2 and 3 are together, depending on what your configuration is), etc.

In this scenario, it would be far less likely that the Stripe payment field would be presented to someone who wasn't ready to purchase. Any "incomplete" records in Stripe would give you a good idea of how many people backed away from completing the purchase, etc.

Stripe also requires that an amount is present before the payment fields are loaded, so if you're attempting to allow users to select or type a payment value, it will go a lot smoother if this interaction happens on a page before the payment fields page. Otherwise, it could potentially error if no value is set, and then force reload (for a couple of seconds) as the user selects a different value, etc. 🙂 

For more information, you can see our Stripe Best Practices documentation: 🙂 
https://docs.solspace.com/craft/freeform/v5/integrations/payments/#best-practices
